main.tf

```diff
@@ -101,6 +103,12 @@ resource "aws_ebs_volume" "volume" {
     tags {
         Name = "DataBoxVolume"
     }
+
+    provisioner "local-exec" {
+	when = "destroy"
+	command = "if [ ${var.create_snapshot} ]; then aws ec2 create-snapshot --volume-id ${aws_ebs_volume.volume.id} --region ${var.aws_region}  --profile ${var.profile} ; fi;"
```

Comment:
@andreagrandi this checks whether anything has been assigned to var.create_snapshot, and if there is any string then it executes the command. This still needs you to set `./databox.sh -c 1 down`, it would be nice if the `1` was not required and you could simply pass `./databox.sh -c down`, but I could not get this to work. Any thoughts?